File: addBorder/addBorder.ts
const addBorder = (picture: string[]): string[] => {

	const longest = picture.reduce( 
        (a, b) => a.length > b.length ? a : b ).length;

    // Extra space for a cleaner look
	const unframed = picture.map(str => str.length < longest 
		? `* ${str}` + ' '.repeat(longest - (str.length-1)) + '*'
		: `* ${str} *`);
    
    const edge = '*'.repeat(longest + 4);
    const pad = `*${' '.repeat(longest + 2)}*`;

    return [ edge, pad, ...unframed, pad, edge ];
}


console.time('addBorder');

console.log(addBorder(['abc', 'ded']).join('\n'));

console.log(addBorder([
        `Let's try a`, 
        'big sentence', 
        'instead of these', 
        'dumb letters.'])
        .join('\n'));

console.log(
	addBorder(['Eat', 'mor', 'chikin']).join('\n'));

console.timeEnd('addBorder');

File: addBorder/add_border.py

from typing import List


def add_border(picture: List[str]) -> List[str]:

    longest = len(max(picture, key=len))

    unframed = [
        '* ' + s + f"{' ' * (longest - (len(s)-1))}*"
        if len(s) < longest
        else f"* {s} *"
        for s in picture
    ]

    edge = '*' * (longest + 4)
    pad = f"*{' ' * (longest + 2)}*"

    return [edge, pad] + unframed + [pad, edge]


print(
    '\n'.join(add_border(['abc', 'ded'])),
    end='\n\n')

print('\n'.join(add_border([
        "Let's try a",
        "big sentence",
        "instead of these",
        "dumb letters."])),
      end='\n\n')

print('\n'.join(
    add_border(['Eat', 'mor', 'chikin'])),
    end='\n\n')
